albatross.ml

```diff
@@ -450,6 +450,37 @@ module Make (S : Tcpip.Stack.V4V6) = struct
                        (Vmm_commands.pp ~verbose:false)
                        cmd TLS.pp_write_error e)
               | Ok tls_flow -> (
+                  let sent = ref false in
+                  let rec send_data () =
+                    push () >>= function
+                    | None ->
+                        if !sent then (
+                          (* send trailing 0 byte chunk *)
+                          let buf = Cstruct.create 4 in
+                          Cstruct.BE.set_uint32 buf 0 0l;
+                          TLS.write tls_flow buf >>= function
+                          | Error _ -> assert false
+                          | Ok () -> Lwt.return_unit)
```

Comment:
As far as I can tell we never set `sent := true`.

Reply:
well spotted (shouldn't there be a OCaml compiler warning)? This also explains why albatross doesn't deploy anything (it waits for the terminating zero-length chunk).

Reply:
It could in this case, but I that the analysis quickly gets hard if the `ref` escapes the scope.